src/utils.ts

```diff
@@ -227,3 +227,13 @@ export const sceneCoordsToViewportCoords = (
 
 export const getGlobalCSSVariable = (name: string) =>
   getComputedStyle(document.documentElement).getPropertyValue(`--${name}`);
+
+export const isRTL = (text: string) => {
+  const ltrChars =
+    "A-Za-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02B8\u0300-\u0590\u0800-\u1FFF" +
+    "\u2C00-\uFB1C\uFDFE-\uFE6F\uFEFD-\uFFFF";
+  const rtlChars = "\u0591-\u07FF\uFB1D-\uFDFD\uFE70-\uFEFC";
+  const rtlDirCheck = new RegExp(`^[^${ltrChars}]*[${rtlChars}]`);
```

Comment:
Why is it necessary to check for negative-LTR chars here?

Reply:
Exactly it doesn't just look if the text contains the RTL chars. But fact, it checks the what is the first directional character it can find,

## There are 2 types of characters:
1- directional: you can infer the direction from the char alone, (ex: English characters or Arabic characters)
2- non-directional: you can not infer the direction from them, (ex: numbers, some punctuation characters like a dot **.**)

## A text direction is defined by the first directional characters that appears in it, examples:
- "English text [maybe some other RTL text here]" is **LTR**
- "123 English text" is **LTR**
- "[Arabic characters] then some English characters" is **RTL**
- "123 [Arabic characters] then some English characters" is **RTL**

That is why we have the negative-LTR, as long as there are RTL characters, with no LTR chars before them, the text then must be RTL

Reply:
Ok, that clears it up, thanks.

Reply:
I've added docs and moved the regexes out of the helper so they're not recreated every call.